src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,184 @@
+  addonIsExtension,
+  addonIsRecommended,
+  experimentEnabled,
+  installStatus,
+}: CouldShowWarningParams) => {
+  return (
+    addonIsExtension &&
+    !addonIsRecommended &&
+    experimentEnabled &&
+    [UNINSTALLED, UNKNOWN].includes(installStatus)
+  );
+};
+
+const addonIsExtension = (addon: AddonType) => {
+  return addon.type === ADDON_TYPE_EXTENSION;
+};
+
+type InternalProps = {|
+  ...Props,
+  ...WithExperimentInjectedProps,
+  _couldShowWarning: typeof couldShowWarning,
+  _log: typeof log,
+  _tracking: typeof tracking,
+  i18n: I18nType,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const EXPERIMENT_CATEGORY_CLICK =
+  'AMO Install Button Warning Experiment - Click';
+export const EXPERIMENT_CATEGORY_SHOW =
+  'AMO Install Button Warning Experiment - Display';
+export const EXPERIMENT_ID = 'installButtonWarning';
+export const INSTALL_WARNING_EXPERIMENT_DIMENSION = 'dimension6';
+export const VARIANT_INCLUDE_WARNING = 'includeWarning';
+export const VARIANT_EXCLUDE_WARNING = 'excludeWarning';
+
+export class InstallWarningBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _couldShowWarning: couldShowWarning,
+    _log: log,
+    _tracking: tracking,
+  };
+
+  maybeSendDisplayTrackingEvent = () => {
+    const {
+      _couldShowWarning,
+      _tracking,
+      addon,
+      experimentEnabled,
+      installStatus,
+      variant,
+    } = this.props;
+
+    if (
+      _couldShowWarning({
+        addonIsExtension: addonIsExtension(addon),
+        addonIsRecommended: addon.is_recommended,
+        experimentEnabled,
+        installStatus,
+      }) &&
+      variant
+    ) {
+      _tracking.sendEvent({
+        action: variant,
+        category: EXPERIMENT_CATEGORY_SHOW,
+        label: addon.name,
+      });
+    }
+  };
+
+  componentDidMount() {
+    const { _log, _tracking, variant } = this.props;
+
+    if (!variant) {
+      _log.debug(`No variant set for experiment "${EXPERIMENT_ID}"`);
+      return;
+    }
+
+    _tracking.setDimension({
+      dimension: INSTALL_WARNING_EXPERIMENT_DIMENSION,
+      value: variant,
+    });
+
+    this.maybeSendDisplayTrackingEvent();
+  }
+
+  componentDidUpdate({ installStatus: oldInstallStatus }: InternalProps) {
+    const { installStatus: newInstallStatus } = this.props;
+
+    if (newInstallStatus !== oldInstallStatus) {
+      this.maybeSendDisplayTrackingEvent();
+    }
+  }
+
+  render() {
+    const {
+      _couldShowWarning,
+      addon,
+      experimentEnabled,
+      i18n,
+      installStatus,
+      variant,
+    } = this.props;
+
+    if (
+      _couldShowWarning({
+        addonIsExtension: addonIsExtension(addon),
+        addonIsRecommended: addon.is_recommended,
+        experimentEnabled,
+        installStatus,
+      }) &&
+      variant === VARIANT_INCLUDE_WARNING
+    ) {
+      return (
+        <Notice
+          actionHref="https://support.mozilla.org/kb/recommended-extensions-program"
```

Comment:
We could probably extract this URL in a constant so that we can use it in the test file.

Reply:
I'm not checking this value in the test file at all, which is why I didn't feel a need to make it a `const` and export it. I'm not testing it because it's just a static value. I can make it a `const` because it is kind of nice to have things like this called out at the top of the file, but it still wouldn't need to be exported.

Reply:
I made this a `const` at the top of the file.